Tools/Scripts/compare-static-analysis-results

```diff
@@ -111,37 +159,39 @@ def compare_project_results(args, archive_path, new_path, project):
     return new_issues_total, new_files_total
 
 
-def create_filtered_results_dir(args, project, issues, category='StaticAnalyzerRegressions'):
-    # Create symlinks to new issues only so that we can run scan-build to generate new index.html files
-    path_to_reports = os.path.abspath(f'{args.build_output}/{category}/{project}/StaticAnalyzerReports')
-    subprocess.run(['mkdir', '-p', path_to_reports])
-    for issue_hash in issues:
-        report = f"report-{issue_hash[:6]}.html"
-        path_to_report = f'{args.build_output}/StaticAnalyzer/{project}/StaticAnalyzerReports/{report}'
-        path_to_report_new = os.path.join(path_to_reports, report)
-        subprocess.run(['ln', '-s', os.path.abspath(path_to_report), path_to_report_new])
-
-    path_to_project = f'{args.build_output}/{category}/{project}'
-    subprocess.run([args.scan_build, '--generate-index-only', os.path.abspath(path_to_project)])
-
-
 def main():
     args = parser()
     new_issues_total = set()
     new_files_total = set()
+    unexpected_passes_total = set()
+    unexpected_failures_total = set()
+    unexpected_issues_total = set()
 
     for project in PROJECTS:
-        archive_path = os.path.abspath(f'{args.archived_dir}/{project}')
-        new_path = os.path.abspath(f'{args.new_dir}/{project}')
         print(f'\n------ {project} ------\n')
-        new_issues, new_files = compare_project_results(args, archive_path, new_path, project)
-        new_issues_total.update(new_issues)
-        new_files_total.update(new_files)
-
+        new_path = os.path.abspath(f'{args.new_dir}')
+        if args.check_expectations:
+            unexpected_failures, unexpected_passes, unexpected_issues = compare_project_results_to_expectations(args, new_path, project)
+            unexpected_failures_total.update(unexpected_failures)
+            unexpected_passes_total.update(unexpected_passes)
+            unexpected_issues_total.update(unexpected_issues)
+        else:
+            archive_path = os.path.abspath(f'{args.archived_dir}/{project}')
+            new_issues, new_files = compare_project_results_by_run(args, archive_path, new_path, project)
+            new_issues_total.update(new_issues)
+            new_files_total.update(new_files)
+
+    print('\n')
     if new_issues_total:
-        print(f'\nTotal new issues: {len(new_issues_total)}')
+        print(f'Total new issues: {len(new_issues_total)}')
     if new_files_total:
         print(f'Total new files: {len(new_files_total)}')
+    if unexpected_failures_total:
+        print(f'Total unexpected failing files: {len(unexpected_failures_total)}')
+    if unexpected_passes_total:
+        print(f'Total unexpected passing files: {len(unexpected_passes_total)}')
+    if unexpected_issues_total:
+        print(f'Total unexpected issues: {len(unexpected_issues_total)}')
```

Comment:
I wonder if this would be more concise as a loop, something like:
```
for description, issues in {
'new issues': new_issues_total,
'unexpected failing files': unexpected_failures_total,
'unexpected passing files': unexpected_passes_total,
'unexpected issues': unexpected_issues_total,
}.items():
if lst:
print(f'Total {description}: {len(lst)}')
```

Reply:
Oh I like this much more, thanks!